packages/protocol/lib/backward/ast-code.ts

```diff
@@ -0,0 +1,425 @@
+// tslint:disable: max-classes-per-file
+import { makeZContract } from '@celo/protocol/lib/backward/internal'
+import {
+  BuildArtifacts,
+  Contract as ZContract
+} from '@openzeppelin/upgrades'
+import ContractAST from '@openzeppelin/upgrades/lib/utils/ContractAST'
+
+export const VISIBILITY_PUBLIC = 'public'
+export const VISIBILITY_EXTERNAL = 'external'
+const CONTRACT_KIND_CONTRACT = 'contract'
+const STORAGE_DEFAULT = 'default'
+
+const OUT_VOID_PARAMETER_STRING = 'void'
+
+// Exported classes
+
+/**
+ * A compatibility report with all the detected changes from two compiled
+ * contract folders.
+ */
+export class ASTCodeCompatibilityReport {
```

Comment:
I did consider this too. As a usage perspective, we are checking the API. But we are extracting the information from the AST. I'm guessing that, if we were to model this in more detail, I'd have an interface to be "CodeComp..Report" or "APICodeComp...Report", but this actual concrete class be "ASTCode...Report". Since this abstraction is not needed, I just left it as "AST...".